mne/decoding/tests/test_ssd.py

```diff
@@ -0,0 +1,240 @@
+    psd_S, _ = psd_array_welch(S[0], sfreq=250, n_fft=250)
+    corr = np.abs(np.corrcoef(psd_out, psd_S)[0, 1])
+    assert np.abs(corr) > 0.95
+    
+    # Check pattern estimation   
+    # Since there is no exact ordering of the recovered patterns 
+    # a pair-wise greedy search will be done
+    err=[]
+        
+    for i in range(n_channels):
+        corr=np.abs(np.corrcoef(ssd.patterns_[i,:].T, A[:, 0])[0, 1])
+        err.append(1-corr)
+        min_err=np.min(err)
+    assert min_err < 0.3 # threshold taken from SSD original paper
+
+    
+def test_ssd_epoched_data():
+    """Test Common Spatial Patterns algorithm on epoched data and compare
+    output when raw data is used."""
+    X, A, S = simulate_data(n_trials=100, n_channels=20,
+                  n_samples=500)
+    sf=250
+    n_channels = X.shape[0]
+    info=create_info(ch_names=n_channels, sfreq=sf, ch_types='eeg')
+    n_components_true=5
+    
+    # Build epochs as sliding windows over the continuous raw file
+    
+    # Epoch length is 1 second
+    X_e = np.reshape(X, (100, 20, 500))
+    
+    # Fit
+    filt_params_signal=dict(l_freq=freqs_sig[0], h_freq=freqs_sig[1],
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin')
+    filt_params_noise=dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin')
+    
+    # ssd on epochs
+    ssd_e= SSD(info, filt_params_signal, filt_params_noise)
+    ssd_e.fit(X_e)
+    
+    # ssd on raw
+    ssd = SSD(info, filt_params_signal, filt_params_noise)
+    ssd.fit(X)
+    
+    # Check if the 5 first 5 components are the same for both
+    _, sorter_spec_e=get_spectral_ratio(ssd_e.transform(X_e), sf, ssd_e)
+    _, sorter_spec=get_spectral_ratio(ssd.transform(X), sf, ssd)
+    assert_array_equal(sorter_spec_e[:n_components_true],
+                              sorter_spec[:n_components_true])   
+            
+@requires_sklearn
+def test_ssd_pipeline():
+    """Test if SSD works in a pipeline."""
+    from sklearn.linear_model import LinearRegression
+    from sklearn.pipeline import Pipeline
+    sf=250
+    info=create_info(ch_names=20, sfreq=sf, ch_types='eeg')
+    
+    filt_params_signal = dict(l_freq=freqs_sig[0], h_freq=freqs_sig[1],
+                                  l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                  fir_design='firwin')
+    filt_params_noise = dict(l_freq=freqs_noise[0], h_freq=freqs_noise[1],
+                                 l_trans_bandwidth=1, h_trans_bandwidth=1,
+                                 fir_design='firwin')
+    ssd = SSD(info, filt_params_signal, filt_params_noise)
+    lm = LinearRegression()
+    pipe = Pipeline([("SSD", ssd), ("LM", lm)])
+    pipe.set_params(SSD__n_components=2)
+    assert (pipe.get_params()["SSD__n_components"] == 2)
```

Comment:
Can you also run the pipeline to make sure it actually does not crash?

Reply:
Added, and worked!